src/mixins/filing-template-mixin.ts

```diff
@@ -418,7 +418,7 @@ export default class FilingTemplateMixin extends DateMixin {
             ? { extension: +this.getBusinessContact.extension }
             : {}
         },
-        expiry: this.getRestoration.expiry || undefined, // can't be null
+        expiry: this.getRestoration.restorationTypeValid ? this.getRestoration.expiry : undefined, // can't be null
```

Comment:
If user saves with invalid number of months and refreshes, handle it.

Reply:
So the user can store an invalid number of months (thus making expiry non-null but not valid)?

If we can't easily fix the underlying problem then this needs a comment (what you wrote above is nice) in the code to explain it.

Reply:
It would as follows:
1. The user saves with invalid number of months.
2. Since the resoration type was invalid, expiry will be undefined.
![no expiry](https://user-images.githubusercontent.com/122301442/222854226-7d6bcb10-ea9b-4b8f-b8dc-c4fcd3220339.PNG)
3. When user refreshes, the selection will still be limited restoration, but the number of months will revert to the original which is 2 years.

Reply:
This is a code smell -- a (small) red flag that something is not right in the design or handling of the data, needing a special work-around to handle it.

What's the root problem we are trying to solve here -- that an invalid number of months was entered? Then one of 2 things should happen instead:
- preferably, we don't save (to store or draft) the invalid number of months; the component can still have a variable set to that value, but don't save it to the store
- or, we save invalid months, but the component indicates that it in an invalid value -- that way, upon resuming a draft, we can still indicate to the user that it is invalid and the filing cannot be submitted

I constantly ask myself, "is this the right place to handle something?" Is it the current code's job to do something with that data (logic), and more generally, should the current code know about this level of information?

In the current situation, it seems to me that this code should only save the data that is in the store, not apply rules to it. So, some other code needs to be responsible for ensuring that the data in the store is valid.

Reply:
I 100% agree. Thing is, no matter how much how tried, I couldn't get the component to validate on load. So I opted to go this path which is not good.

I think on Friday my mind was so fried, I couldn't think straight. I just figured a small change to let my component emit a validation on load. This change is going to be in the sub-component so the current code doesn't need to worry about that.

The way it's going to work, I'll let the user save an invalid number of months. When the user saves and refreshes, the component is going to be invalid. Just like we do with the other components.

Reply:
This change was reverted. Closing this conversation.